### ISSUE TITLE
Unknown, possibly peripheral-related crash

### ISSUE THREAD

**Issue by KSSBrawl** [TerryCavanagh/VVVVVV#1153]
A friend of mine is having an issue on a Windows 11 laptop (specifically [this one](https://www.bestbuy.com/site/hp-omen-16-1-qhd-gaming-laptop-intel-core-i7-16gb-memory-nvidia-geforce-rtx-3070-1tb-ssd-ceramic-white/6479084.p?skuId=6479084)). They were playing the game with a 2160x1440 monitor plugged in and using fullscreen and the game ran fine, but once they disconnect the monitor it consistently crashes with an access violation (0xc0000005) error code. Additionally, although they have the scaling mode set to integer scaling, the game would start stretched and crash before it actually used integer scaling. They do not _currently_ have the laptop on their person (it is at their dorm) but come Monday if needed they can record a video of the crash.

**Comment by Daaaav** [comment]
Myself I cannot get this to happen on either:
- Steam 2.4(.0) on Windows 10
- Steam 2.4(.0) on Linux
- upstream 2.4.1 on Linux

In all cases, I tried both disconnecting the external monitor while the game is running, as well as disconnecting it between runs. The game would always fall back to the main display.

What did happen specifically on Windows was that if I disconnect the external monitor while the game is running, the game display freezes (after correctly moving the window to the main display) but sound and input was still working.

**Comment by flibitijibibo** [comment]
A laptop with NV hardware, eh...

If they rename VVVVVV.exe to doom3.exe does that fix it?

**Comment by Buggem** [comment]
Judging from the error being specifically based on monitor choice (eg builtin versus external), I'd say it could be an issue with either upstream SDL2 or your friend's drivers (if your friend has installed sketchy/atypical drivers).

But it really depends whether other SDL2 games have this problem on the same device.

The only thing leading to VVVVVV itself is the scaling, but even that is mostly handled by SDL (the proportions being an SDL2_Rect). I, being a Linux user, don't know much about debugging in Windows, but I'm sure that there are many programs that can help you trace it back to SDL2 or VVVVVV (or even your friends drivers)

**Comment by KSSBrawl** [comment]
I totally forgot about this issue honestly. It could totally be the drivers, and also I can ask them to try renaming to doom3.exe, which I assume is to try and trick whatever performance optimizer is installed?

**Comment by Buggem** [comment]
@KSSBrawl I really can't find any evidence that changing a game.exe to doom3.exe (presumably DOOM 3 by id Software) would help. Try getting a driver update from the official nvidia website and see if it works.

**Comment by flibitijibibo** [comment]
> I totally forgot about this issue honestly. It could totally be the drivers, and also I can ask them to try renaming to doom3.exe, which I assume is to try and trick whatever performance optimizer is installed?

Yeah, this is usually to get Optimus to do the right thing, since it might be defaulting to integrated graphics (usually not a big deal for VVV but if it fixes the bug...)
